src/libYARP_os/src/yarp/os/Type.cpp

```diff
@@ -108,6 +109,27 @@ Type::~Type()
     delete mPriv;
 }
 
+std::string Type::getVersion() const
+{
+    return mPriv->version;
+}
+
+size_t Type::getMajorVersion() const
+{
+    unsigned int major = 0;
+    unsigned int minor = 0;
+    int ret = sscanf(mPriv->version.c_str(),"%u.%u", &major, &minor);
+    return major;
+}
+
+size_t Type::getMinorVersion() const
+{
+    unsigned int major = 0;
+    unsigned int minor = 0;
+    int ret = sscanf(mPriv->version.c_str(), "%d.%d", &major, &minor);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [%d in format string (no. 2) requires 'int *' but the argument type is 'unsigned int *'.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=5175088)